homeassistant/components/sql/config_flow.py

```diff
@@ -139,6 +179,10 @@ async def async_step_user(
                 options[CONF_UNIT_OF_MEASUREMENT] = uom
             if value_template := user_input.get(CONF_VALUE_TEMPLATE):
                 options[CONF_VALUE_TEMPLATE] = value_template
+            if (device_class := user_input.get(CONF_DEVICE_CLASS)) != NONE_SENTINEL:
```

Comment:
As you now have a default no need to use `.get` as it always has a value.
Also this change does not consider the scenario when you go from a `device_class` to `"None"` to clean it off. It would just always keep the previous `device_class`

Reply:
Could probably be worth to make a small function to check this as you have the same "issue" in both config and options flow

Reply:
> Also this change does not consider the scenario when you go from a `device_class` to `"None"` to clean it off. It would just always keep the previous `device_class`

This cleaning already happens, the `options` dictionary is redefined at each manipulation, so the current logic does not include the previous value.